### ISSUE TITLE
Support for Koa?

### ISSUE THREAD

**Issue by bishtawi** [jhollingworth/marty-express#5]
Hi,
Any plans to support the [koa web framework](https://github.com/koajs/koa)?

**Comment by jhollingworth** [comment]
No plans right now to specifically implement Koa however I'm planning on abstracting the the internals so it's easy to add support for other web frameworks (e.g. Hapi)

Out of interest, beyond generators are there many differences with express?

**Comment by bishtawi** [comment]
The biggest change was to replace callbacks with generators. They also combined the res and req objects into a single context (the this object) and moved a lot of things to middleware. There are others, but thats really the big obvious differences. I modified your source to make it work with Koa. All I did was turn it into a generator, replace res and req with the koa contex, replace the res.render function with React.renderToStaticMarkup and added support for document title and 404 error. It works well enough for me to continue with my project.

**Comment by tychotatitscheff** [comment]
I would add a big _+1_.
I can open a repo.

@bishtawi would you like to share some code. I was about to work on it but if you already done this start it would be nice.

**Comment by bishtawi** [comment]
Yeah, thats a good idea. I'll throw up the code I have up onto a repo later tonight. I didnt bother fixing the tests yet.

**Comment by bishtawi** [comment]
Alright, marty-koa is available [here](https://github.com/bishtawi/marty-koa).
Let me know if you have any problems.

**Comment by tychotatitscheff** [comment]
Wow. Nice work indeed.
I would test it soon and open up issues if I have any problems.
